assets/dev/js/frontend/handlers/container/grid-outline.js

```diff
@@ -0,0 +1,229 @@
+
+		return {
+			outlineParentContainer: null,
+			gridOutline: this.findElement( selectors.gridOutline ),
+			directChildGridOverlay: this.findElement( selectors.directGridOverlay ),
+		};
+	}
+
+	onInit() {
+		super.onInit();
+
+		this.initLayoutOverlay();
+	}
+
+	bindEvents() {
+		elementorFrontend.elements.$window.on( 'resize', this.onDeviceModeChange.bind( this ) );
+		this.addChildLifeCycleEventListeners();
+	}
+
+	unbindEvents() {
+		this.removeChildLifeCycleEventListeners();
+		elementorFrontend.elements.$window.off( 'resize', this.onDeviceModeChange.bind( this ) );
+	}
+
+	initLayoutOverlay() {
+		this.getContainer();
+
+		if ( ! this.elements.outlineParentContainer ) {
+			return;
+		}
+
+		this.removeExistingOverlay();
+		this.createOverlayContainer();
+		this.createOverlayItems();
+	}
+
+	getContainer() {
+		const { grid_outline: gridOutline } = this.getElementSettings();
+
+		this.elements.outlineParentContainer = null;
+
+		if ( gridOutline && '' !== gridOutline ) {
+			return this.getCorrectContainer();
+		}
+	}
+
+	getCorrectContainer() {
+		const container = this.$element[ 0 ],
+			getDefaultSettings = this.getDefaultSettings(),
+			{ selectors: { boxedContainer } } = getDefaultSettings;
+
+		this.elements.outlineParentContainer = container.querySelector( boxedContainer ) || container;
+	}
+
+	removeExistingOverlay() {
+		const { gridOutline } = this.elements;
+
+		gridOutline?.remove();
+	}
+
+	createOverlayContainer() {
+		const { outlineParentContainer } = this.elements,
+			{ classes: { outline } } = this.getDefaultSettings(),
+			gridOutline = document.createElement( 'div' );
+
+		gridOutline.classList.add( outline );
+		outlineParentContainer.appendChild( gridOutline );
+
+		this.elements.gridOutline = gridOutline;
+
+		this.getGridDimensions();
+		this.setGridOutlineDimensions();
+	}
+
+	createOverlayItems() {
+		const { gridOutline } = this.elements,
+			{ classes: { outlineItem } } = this.getDefaultSettings(),
+			{ rows, columns } = this.getGridDimensions(),
+			numberOfItems = rows.length * columns.length;
+
+		for ( let i = 0; i < numberOfItems; i++ ) {
+			const gridOutlineItem = document.createElement( 'div' );
+
+			gridOutlineItem.classList.add( outlineItem );
+			gridOutline.appendChild( gridOutlineItem );
+		}
+	}
+
+	/**
+	 * Get the grid dimensions for the current device.
+	 *
+	 * @return { { columns: { value, length }, rows: { value, length } } }
+	 */
+	getGridDimensions() {
+		const currentDevice = elementor.channels.deviceMode.request( 'currentMode' );
+
+		return {
+			rows: this.getControlValues( 'grid_rows_grid', currentDevice, 'grid-template-rows' ) || 1,
+			columns: this.getControlValues( 'grid_columns_grid', currentDevice, 'grid-template-columns' ) || 1,
+		};
+	}
+
+	setGridOutlineDimensions() {
+		const { gridOutline } = this.elements,
+			{ rows, columns } = this.getGridDimensions();
+
+		gridOutline.style.gridTemplateColumns = columns.value;
+		gridOutline.style.gridTemplateRows = rows.value;
```

Comment:
@davseve Are you overriding the control's `grid-template-columns` and `grid-template-rows` here inline?

Reply:
Yes, that is what I do.

Reply:
@davseve Can you maybe just add a comment somewhere regarding this issue? The fact that we need to cater for users adding more content into the widget than what the initial rows/columns are. Because this resulted in some complexity in the code and it is not really clear currently. Like I needed you to physically explain it to me.